src/test/java/org/broadinstitute/consent/http/resources/DatasetResourceTest.java

```diff
@@ -673,6 +674,46 @@ public void testGetDatasetNotFound() {
         assertEquals(404, response.getStatus());
     }
 
+    @Test
+    public void testGetDatasets() {
+        Dataset ds1 = new Dataset();
+        ds1.setDataSetId(1);
+        Dataset ds2 = new Dataset();
+        ds2.setDataSetId(2);
+        Dataset ds3 = new Dataset();
+        ds3.setDataSetId(3);
+
+        when(datasetService.getDatasets(List.of(1,2,3))).thenReturn(List.of(
+                ds1,
+                ds2,
+                ds3
+        ));
+
+        initResource();
+        Response response = resource.getDatasets(List.of(1,2,3));
+        assertEquals(200, response.getStatus());
+        assertEquals(List.of(ds1,ds2,ds3), response.getEntity());
+    }
+
+    @Test
+    public void testGetDatasetsNotFound() {
+        Dataset ds1 = new Dataset();
+        ds1.setDataSetId(1);
+        Dataset ds3 = new Dataset();
+        ds3.setDataSetId(3);
+
+
+        when(datasetService.getDatasets(List.of(1,2,3,4))).thenReturn(List.of(
+                ds1,
+                ds3
+        ));
+
+        initResource();
+        Response response = resource.getDatasets(List.of(1,2,3,4));
+        assertEquals(404, response.getStatus());
+        assertEquals("Could not find datasets with ids: 2,4", ((Error)response.getEntity()).getMessage());
```

Comment:
is there a more efficient way to check error message?

Reply:
IMHO, message testing is not very helpful and often brittle so I discourage it. But a direct answer is no, reading the entity is the only way to get the content out of the response.

Reply:
One thing you could do is test for the existence/non-existence of the IDs in the string instead of looking at complete string equality.

Reply:
ahh good idea, just implemented I believe a decent way of doing this.